cypress/integration/learn/challenge-hot-keys.js

```diff
@@ -0,0 +1,73 @@
+const selectors = {
+  instructions: '.challenge-instructions',
+  instructionsPanel: '.instructions-panel',
+  editorContainer: '.monaco-editor',
+  console: '.output-text'
+};
+
+const links = {
+  link1:
+    '/learn/responsive-web-design/basic-html-and-html5/say-hello-to-html-elements',
+  link2:
+    '/learn/responsive-web-design/basic-html-and-html5/headline-with-the-h2-element',
+  link3:
+    '/learn/responsive-web-design/responsive-web-design-projects/build-a-tribute-page',
+  link4:
+    '/learn/responsive-web-design/responsive-web-design-projects/build-a-survey-form',
+  videoLink1:
+    '/learn/scientific-computing-with-python/python-for-everybody/introduction-why-program',
+  videoLink2:
+    'learn/scientific-computing-with-python/python-for-everybody/introduction-hardware-architecture'
+};
+
+describe('The hotkeys should work correctly', () => {
+  beforeEach(() => {
+    cy.visit(links.link1);
+  });
+
+  it('should be possible to navigate to the next challenge/projects and previous', () => {
+    cy.focused().type('{esc}');
+    cy.focused().type('n');
+    cy.url().should('include', links.link2);
+    cy.focused().type('{esc}');
+    cy.focused().type('p');
+    cy.url().should('include', links.link1);
+    cy.visit(links.link3);
+    cy.focused().type('{esc}').type('n');
+    cy.url().should('include', links.link4);
+    cy.focused().type('{esc}').type('p');
+    cy.url().should('include', links.link3);
+  });
```

Comment:
@ojeytonwilliams All I mean is lines 29-31 test the same thing as lines 36-37, and 32-34 the same as 38-39.

Unless I am missing how chaining `.type().type()` is different to `cy.focused().type(); cy.focused().type()`

Reply:
Ah, I'm with you.  The point is that `.type().type()` types into the same element twice.  If you do `cy.focused().type('{esc}'); cy.focused().type('n')` then you 

1. get the currently focused element
2. press escape which changes the focus 
3. get the newly focused element
4. press 'n'

If you don't switch focus, you just end up typing in the editor.

Reply:
That does clear it up. So, we still want to leave all of these tests to check for both focusable environments?

Reply:
What do you mean by 'focusable environments'?

If you mean the different types of challenges, we probably should have an extra tests for backend to make sure we can navigate everywhere.

Reply:
Yes, I mean the different types of challenges.

Reply:
Cool, they're all tested now.